### ISSUE TITLE
Cannot restore renv due to missing binary and source in Posit Package Manager

### ISSUE THREAD

**Issue by lhjohn** [OHDSI/Strategus#86]
I posted this in the https://github.com/OHDSI/CohortGeneratorModule/issues/36, but it may be an issue that can better be resolved from Strategus:

By default `renv` uses Posit Package Manager (PPM) to find binaries for the `renv` environment. However, PPM does not provide binary or source for some packages:
```
# Downloading packages -------------------------------------------------------
Warning: failed to find binary for 'MASS 7.3-58.2' in package repositories
Warning: failed to find source for 'MASS 7.3-58.2' in package repositories
Error: failed to retrieve package 'MASS@7.3-58.2'
Traceback (most recent calls last):
8: renv::restore(prompt = FALSE)
7: renv_restore_run_actions(project, diff, current, lockfile, rebuild)
6: retrieve(packages)
5: handler(package, renv_retrieve_impl(package))
4: renv_retrieve_impl(package)
3: renv_retrieve_repos(record)
2: stopf("failed to retrieve package '%s'", renv_record_format_remote(record))
1: stop(sprintf(fmt, ...), call. = call.)
Execution halted
```

Is there any way we can set another mirror? So far no luck with using environmental variables as follows:
- `RENV_CONFIG_PPM_ENABLED=FALSE`
- `RENV_CONFIG_PPM_URL="http://lib.stat.cmu.edu/R/CRAN/"`

What worked to some extend (populate `renv`, but not working to run study yet) is the following inside of the modules:
`options("repos" = "https://cran.r-project.org/")`, then `renv::restore(repos = getOption("repos"))`

**Comment by egillax** [comment]
I don't think this is because the source/binary is not available in PPM. For example the source of the package in the issue is here: https://packagemanager.posit.co/cran/latest/src/contrib/Archive/MASS/MASS_7.3-58.2.tar.gz , and can be downloaded.

I've also been running into this error. I'm wondering if we're running into some kind of api rate limitation of PPM ?

**Comment by anthonysena** [comment]
@lhjohn - just wanted to check that you are using the code on the `develop` branch for this? Assuming you are, then I would expect that part of this may be related to #81 since when we use `renv::run` (see below) it does not preserve the options that are set outside of the .Renviron (again assuming you are setting those environment variables via `Sys.setenv`). Let me know if my assumptions hold. Also, @egillax raises an interesting point - do you know if such a rate limit exists and how it is enforced? 

(Reference to the code in Strategus that performs the module instantiation via an `renv::run()` so the module is restore in its own context)

https://github.com/OHDSI/Strategus/blob/25499ec3b6f03ed95aaec25facf4bc5692cd30c1/R/ModuleInstantiation.R#L194-L214

**Comment by egillax** [comment]
Hi @anthonysena,

Is there a reason that in the latest modules you are using [posit package manager](https://github.com/OHDSI/CohortGeneratorModule/blob/main/renv.lock#L7) instead of CRAN as the repo in the lockfile? This is at least causing me and @lhjohn these sporadic errors during module instantiation where it claims not to find certain packages.

**Comment by lhjohn** [comment]
What @egillax mentions is likely the problem. I believe `renv` started to use posit package manager by default as from v1.0.0. Can we instead keep using the standard CRAN mirror until that new PPM mirror becomes more "stable"?

**Comment by anthonysena** [comment]
Per @egillax 
> Is there a reason that in the latest modules you are using [posit package manager](https://github.com/OHDSI/CohortGeneratorModule/blob/main/renv.lock#L7) instead of CRAN as the repo in the lockfile?

Yes, this is the default when initializing a new project using `renv` as shown from starting a fresh project (renv.lock below for ref). I'd prefer not to override this in the lock files since this is the default behavior of `renv`. 

```json
{
  "R": {
    "Version": "4.2.3",
    "Repositories": [
      {
        "Name": "CRAN",
        "URL": "https://packagemanager.posit.co/cran/latest"
      }
    ]
  },
  "Packages": {
    "renv": {
      "Package": "renv",
      "Version": "1.0.2",
      "Source": "Repository",
      "Repository": "CRAN",
      "Requirements": [
        "utils"
      ],
      "Hash": "4b22ac016fe54028b88d0c68badbd061"
    }
  }
}
```

Per @lhjohn:

> Can we instead keep using the standard CRAN mirror until that new PPM mirror becomes more "stable"?

I think I'd like to modify Strategus to support the passing of the environment's info into the `renv::run` command so that you can make changes like the ones you mentioned earlier in this issue:

```
RENV_CONFIG_PPM_ENABLED=FALSE
RENV_CONFIG_PPM_URL="http://lib.stat.cmu.edu/R/CRAN/"
```

And then you don't need to modify the modules to get it to work in your environment. Let me try this on a branch and will let you know when its ready.

**Comment by anthonysena** [comment]
OK - I think I've got a patch for this if you want to give it a try? The code is on the branch `renv-option-passing`.

The code is a little hacky in that it will attempt to carry-over any of the "renv.*" configuration options using the `options()` set in your environment. Here is a link to the full list of renv options: https://rstudio.github.io/renv/reference/config.html.

As a test, I was able to use the `renv.config.repos.override` option to specify a new URL for downloading resources as shown below. I also tried setting this to an invalid URL to verify that it was indeed using this value and it failed as expected.

```r
analysisSpecifications <- ParallelLogger::loadSettingsFromJson(
  fileName = system.file("testdata/unitTestAnalysisSpecification.json",
                         package = "Strategus"
  )
)

options(renv.config.repos.override = c("http://lib.stat.cmu.edu/R/CRAN/"))
Strategus::ensureAllModulesInstantiated(
  analysisSpecifications = analysisSpecifications
)
```

If this meets your needs, I'll add this this to the `develop` branch as there was also another bug I discovered while I was making these updates.
